### PR TITLE
Added RoleDao, UserDao, User Service, Role Service, DataProcessingExc…

### PR DESCRIPTION
**Pull request by Alekearl**
…eption and models Role and User

### CODE REVIEW

**Review by engineer027 — APPROVED** [review]

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -1 +0,0 @@
-
```
Comment (KseniiaMakarova) [inline src/main/resources/application.properties:1]: let’s configure access to h2 database through the console in `application.properties`

File: src/main/java/com/bank/project/bank/dao/RoleDao.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package com.bank.project.bank.dao;
+
+import com.bank.project.bank.model.Role;
+import org.springframework.data.jpa.repository.JpaRepository;
+import java.util.Optional;
+
+public interface RoleDao extends JpaRepository<Role, Long> {
+    Optional<Role> getByRoleName(String roleName);
```
Comment (KseniiaMakarova) [inline src/main/java/com/bank/project/bank/dao/RoleDao.java]:
```suggestion
    Optional<Role> getByRoleName(Role.RoleName roleName);
```

File: src/main/java/com/bank/project/bank/dao/UserDao.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package com.bank.project.bank.dao;
+
+import com.bank.project.bank.model.User;
+import org.springframework.data.jpa.repository.JpaRepository;
+import java.util.Optional;
+
+public interface UserDao extends JpaRepository<User, Long> {
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/bank/project/bank/dao/UserDao.java]: redundant empty line

File: src/main/java/com/bank/project/bank/dao/UserDao.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+    Optional<User> findByPhoneNumber(String phoneNumber);
```
Comment (KseniiaMakarova) [inline src/main/java/com/bank/project/bank/dao/UserDao.java:11]: see checklist - in this method and also in `getById()` - we should join fetch roles as well

File: src/main/java/com/bank/project/bank/exception/DataProcessingException.java [hunk 5]
```diff
@@ -0,0 +1,6 @@
+package com.bank.project.bank.exception;
+
+public class DataProcessingException extends RuntimeException {
+    public DataProcessingException(String message) {
+    }
+}
```
Comment (KseniiaMakarova) [inline src/main/java/com/bank/project/bank/exception/DataProcessingException.java]:
```suggestion
public class DataProcessingException extends RuntimeException {
    public DataProcessingException(String message) {
        super(message);
    }
}
```

File: src/main/java/com/bank/project/bank/service/UserService.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
+package com.bank.project.bank.service;
+
+import com.bank.project.bank.model.User;
+
+public interface UserService {
+    User create(User user);
+
+    User update(User user);
```
Comment (KseniiaMakarova) [inline src/main/java/com/bank/project/bank/service/UserService.java]: see checklist

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
I noticed that you have a lot of redundant starters in your pom.xml

currently you will need only parent starter, web starter, jpa starter, h2 and lombok dependencies

File: src/main/resources/application.properties (hunk 1 above)
Comment (Alekearl) [reply src/main/resources/application.properties:1]: Ok

File: src/main/java/com/bank/project/bank/dao/UserDao.java (hunk 3 above)
Comment (Alekearl) [reply src/main/java/com/bank/project/bank/dao/UserDao.java]: removed

File: src/main/java/com/bank/project/bank/dao/UserDao.java (hunk 4 above)
Comment (Alekearl) [reply src/main/java/com/bank/project/bank/dao/UserDao.java:11]: Ok

File: src/main/java/com/bank/project/bank/service/UserService.java (hunk 6 above)
Comment (Alekearl) [reply src/main/java/com/bank/project/bank/service/UserService.java]: Ok

File: pom.xml [hunk 7]
```diff
@@ -45,11 +41,6 @@
             <groupId>org.springframework.session</groupId>
             <artifactId>spring-session-core</artifactId>
```
Comment (KseniiaMakarova) [inline pom.xml]: you don't need this as well

File: pom.xml [hunk 8]
```diff
@@ -29,10 +29,6 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-freemarker</artifactId>
```
Comment (KseniiaMakarova) [inline pom.xml]: same

File: pom.xml [hunk 9]
```diff
@@ -29,10 +29,6 @@
             <groupId>org.springframework.boot</groupId>
```
Comment (KseniiaMakarova) [inline pom.xml:33]: same for jdbc starter

File: src/main/resources/application.properties [hunk 10]
```diff
@@ -0,0 +1,10 @@
+spring.datasource.url=jdbc:h2:mem:testdb
+spring.datasource.driverClassName=org.h2.Driver
+spring.datasource.username=user
+spring.datasource.password=1234
+spring.jpa.database-platform=org.hibernate.dialect.H2Dialect
+spring.h2.console.enabled=true
+spring.h2.console.path=/h2-console
+spring.h2.console.settings.trace=false
+spring.h2.console.settings.web-allow-others=false
+server.port=8080
```
Comment (KseniiaMakarova) [inline src/main/resources/application.properties]: add empty line

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
should we add checkstyle plugin?

File: pom.xml (hunk 7 above)
Comment (Alekearl) [reply pom.xml]: Ok

File: pom.xml (hunk 8 above)
Comment (Alekearl) [reply pom.xml]: Ok

File: pom.xml (hunk 9 above)
Comment (Alekearl) [reply pom.xml:33]: Ok

File: src/main/resources/application.properties (hunk 10 above)
Comment (Alekearl) [reply src/main/resources/application.properties]: Ok

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**